software/main.py

```diff
@@ -0,0 +1,97 @@
+from ast import Break
+from asyncio import constants
+from concurrent.futures import process
+from pickle import TRUE
+import image_processor
+import camera
+#import motion
+import cv2
+import time
+from math import *
+import numpy as np
+import movement
+
+spin = 10
+
+
+
+
+
+def main_loop():
+    debug=True
+    
+    cam=camera.RealsenseCamera(exposure=100)
+    processor = image_processor.ImageProcessor(cam, debug=debug, color_config = "colors/colors.pkl")
+    global robot
+    robot=movement.OmniRobot()
+    
+    processor.start()
+    start=time.time()
+    fps=0
+    frame=0
+    frame_cnt=0
+    
+    try:
+        while True:
+            processedData= processor.process_frame(aligned_depth=False)
+            frame_cnt+=1
+            frame+=1
+            #robot.try_motors()
+            #break
+            if frame%30==0:
+                frame = 0
+                end = time.time()
+                fps = 30/(end-start)
+                start = end
+                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                print(processedData.balls)  
+        
+                if (len(processedData.balls)!=0):

```

Comment:
Check https://ut-robotics.github.io/picr22-home/basketball_robot_guide/software/review_summary.html#_use_state_machine_for_game_logic and https://ut-robotics.github.io/picr22-home/basketball_robot_guide/software/review_summary.html#_use_enums

Reply:
There is no enum yet but there is state machine now.

Reply:
Yeah, it is better but a Enum is good to use in this case because there is a defined set of states that need to be handled. With a Enum, compared to a string, you define the set in a singular way that increases code robustness.